### PR TITLE
 Do not allow super() without parameters in generator expressions #2310

### LINKED ISSUE #2310: Do not allow `super()` without parameters in generator expressions
This bug: https://bugs.python.org/issue46175

We need to require `super(cls, self)` in this case.

### PR DESCRIPTION
**Pull request by kitsiosvas**
# I have made things!

## Checklist

- [ x] I have double checked that there are no unrelated changes in this pull request (old patches, accidental config files, etc)
- [ ] I have created at least one test case for the changes I have made
- [ ] I have updated the documentation for the changes I have made
- [ ] I have added my changes to the `CHANGELOG.md`

## Related issues

 Closes #2310 

🙏 Please, if you or your company is finding wemake-python-styleguide valuable, help us sustain the project by sponsoring it transparently on https://opencollective.com/wemake-python-styleguide. As a thank you, your profile/company logo will be added to our main README which receives hundreds of unique visitors per day.

### CODE REVIEW

File: wemake_python_styleguide/visitors/ast/classes.py [hunk 1]
```diff
@@ -236,6 +236,18 @@ def _get_call_stmt_of_useless_method(
...
+        self,
+        call: ast.Call,
+    ) -> None:
+        function_name = functions.given_function_called(call, {'super'})
+        if function_name != 'super':
+            return
+        elif len(call.args) == 2:
+            return
+        elif walk.get_closest_parent(call, ast.GeneratorExp):
+            WrongMethodVisitor(call, text=function_name)
```
Comment (sobolevn) [inline wemake_python_styleguide/visitors/ast/classes.py:249]: Use `self.add_violation`

File: wemake_python_styleguide/visitors/ast/classes.py [hunk 2]
```diff
@@ -236,6 +236,18 @@ def _get_call_stmt_of_useless_method(
             return stmt.value
         return None
 
+    def _check_super_in_generator_expr(
+        self,
+        call: ast.Call,
+    ) -> None:
+        function_name = functions.given_function_called(call, {'super'})
+        if function_name != 'super':
```
Comment (sobolevn) [inline wemake_python_styleguide/visitors/ast/classes.py:244]:
```suggestion
        if not function_name:
```

**Review by sobolevn — CHANGES_REQUESTED** [review]
Thanks! 👍

**Comment by kitsiosvas** [comment]
Hello, I was wondering, is there an issue with my contribution on this pull request? I see it being frozen for some days, and I think I can help if there is anything I can do. 
Thank  you

**Comment by sobolevn** [comment]
@kitsiosvas no worries, you are doing great! I will help you polishing this PR when I will release `0.16.1` (it might take from one to several month). 

You will get full credit for this feature! 🎉 

Thank you for your help!

**Pull request closed without merging**